programs/session-manager/src/token/approve.rs

```diff
@@ -0,0 +1,95 @@
+use crate::error::SessionManagerError;
+use crate::{StartSession, SESSION_SETTER_SEED};
+use anchor_lang::prelude::*;
+use anchor_spl::token::approve_checked;
+use anchor_spl::{
+    associated_token::get_associated_token_address,
+    token::{spl_token::try_ui_amount_into_amount, ApproveChecked, Mint},
+};
+use mpl_token_metadata::accounts::Metadata;
+use solana_intents::SymbolOrMint;
+
+impl<'info> StartSession<'info> {
+    /// Delegate token accounts to the session key.
+    /// Signing an intent with the symbol "SOL" means delegating your token account for a token who has metadata symbol "SOL".
+    /// Although there can be multiple tokens with the same symbol, the worst case scenario is that you're delegating the token with the most value among them, which is probably what you want.
+    pub fn approve_tokens(
+        &self,
+        accounts: &[AccountInfo<'info>],
+        tokens: &[(SymbolOrMint, String)],
+        user: &Pubkey,
+        session_setter_bump: u8,
+    ) -> Result<Vec<Pubkey>> {
+        let mut approved_mints = vec![];
+        let mut accounts_iter = accounts.iter();
+        for (symbol_or_mint, amount) in tokens.iter() {
+            let (user_account, mint_account) = match symbol_or_mint {
+                SymbolOrMint::Symbol(symbol) => {
+                    let user_account = accounts_iter
+                        .next()
+                        .ok_or(error!(SessionManagerError::MissingAccount))?;
+                    let mint_account = accounts_iter
+                        .next()
+                        .ok_or(error!(SessionManagerError::MissingAccount))?;
+                    let metadata_account = accounts_iter
+                        .next()
+                        .ok_or(error!(SessionManagerError::MissingAccount))?;
+
+                    require_eq!(
+                        metadata_account.key(),
+                        Metadata::find_pda(&mint_account.key()).0,
+                        SessionManagerError::MetadataMismatch
+                    );
+                    let metadata = Metadata::try_from(metadata_account)?;
+                    require_eq!(
+                        &metadata.symbol,
+                        &format!("{symbol:\0<10}"),
+                        SessionManagerError::SymbolMismatch
```

Comment:
i haven't really seen this code before, so forgive the question on this incumbent part of the code: why would we allow anyone to specify a string symbol as opposed to explicitly providing the mint address?